Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift

```diff
@@ -30,12 +32,18 @@ public struct FirstWhereRule: CallPairRule, OptInRule, ConfigurationProviderRule
     )
 
     public func validate(file: File) -> [StyleViolation] {
-        return validate(file: file,
-                        pattern: "[\\}\\)]\\s*\\.first",
-                        patternSyntaxKinds: [.identifier],
-                        callNameSuffix: ".filter",
-                        severity: configuration.severity) { dictionary in
-            if !dictionary.substructure.isEmpty {
+        return validate(
+            file: file,
+            pattern: "[\\}\\)]\\s*\\.first",
+            patternSyntaxKinds: [.identifier],
+            callNameSuffix: ".filter",
+            severity: configuration.severity
+        ) { dictionary in
+            if
+                !dictionary.substructure.isEmpty &&
+                dictionary.substructure.last!["key.kind"] as? String != "source.lang.swift.expr.argument" &&
```

Comment:
1. Can you remove the force unwraps?
2. We have extensions on `Dictionary` to get access to `key.kind` (`dictionary.kind`) and `key.name` (`dictionary.name`)
3. You can use the value from `SwiftExpressionKind`: `dictionary.substructure.last.kind.flatMap(SwiftExpressionKind.init) != .argument`

Reply:
Done.